sdk/javascript/test/utils/converter_spec.js

```diff
@@ -181,4 +181,144 @@ describe('converter', () => {
 		addTryParseFailureTest('cannot parse negative decimal string', '-14952');
 		addTryParseFailureTest('cannot parse arbitrary string', 'catapult');
 	});
+
+	describe('bytesToBigInt', () => {
+		const canParseUnsigned = (array, size, expectedValue) => {
+			// Act:
+			const actual = converter.bytesToBigInt(new Uint8Array(array), size);
+
+			// Assert:
+			expect(actual).to.equal(expectedValue);
+		};
+
+		const canParseSigned = (array, size, expectedValue) => {
+			// Act:
+			const actual = converter.bytesToBigInt(new Uint8Array(array), size, true);
+
+			// Assert:
+			expect(actual).to.equal(expectedValue);
+		};
+
+		it('can parse numbers with common sizes', () => {
+			const array = [0xEE, 0xCC, 0xDD, 0x99, 0x33, 0x55, 0xAA, 0xFE];
+
+			canParseUnsigned(array, 8, BigInt('0xFEAA553399DDCCEE'));
+			canParseUnsigned(array, 4, BigInt('0x99DDCCEE'));
+			canParseUnsigned(array, 2, BigInt('0xCCEE'));
+			canParseUnsigned(array, 1, BigInt('0xEE'));
+		});
+
+		it('cannot parse too short array', () => {
+			// Act:
+			expect(() => converter.bytesToBigInt(new Uint8Array([0xaa, 0x55, 0xcc, 0x33, 0xfe, 0xdc, 0xba, 0x98]), 9)).to.throw();
+		});
+
+		it('can parse data as signed numbers', () => {
+			// Arrange: data in both arrays is aranged so there are positive and negative numbers
+			const positiveArray = [0x7E, 0x7C, 0xDD, 0x79, 0x33, 0x55, 0xAA, 0x7E];
+			const negativeArray = [0xEE, 0xCC, 0xDD, 0x99, 0x33, 0x55, 0xAA, 0xFE];
+
+			canParseSigned(positiveArray, 8, BigInt('0x7EAA553379DD7C7E'));
+			canParseSigned(negativeArray, 8, -BigInt(' 0x155aacc66223312'));
```

Comment:
capitals + spurious leading space in string

Reply:
hmm, that spurious space waas there earlier to align it nicely, but now it doesn't make sense :/